direct.c

```diff
@@ -501,6 +502,8 @@ rr_data_t direct_request(void *cdata, rr_data_const_t request) {
 					goto bailout;
 				}
 			}
+
+			++loop;
```

Comment:
Does it even make sense to use loops for this?

Reply:
Probably it does not, it could be made sequential because every code block is wrapped between `if (loop == 0)` or `if (loop == 1)`.

Here SonarCloud complains for the "for loop" because if you read `for (loop = 0; loop < 2; ++loop)` you expect that the `loop` variable increases after each iteration, but we have at line 433 `loop = 0;`. Again this is confusing. If you use a "while" iteration you expect that the variable `loop` will change somewhere in that code block.

Reply:
I would say maybe the loops had best be removed entirely then. I'm afraid the current situation is still somewhat hard to read. But it's up to you :)

Reply:
Yes, I agree with you. The loop is there because the operation of reading headers must be performed twice. First for the request, then for the response. In `direct.c` there is also a `goto shortcut` that is even more confusing. I don't like using `gotos` in programs, it is very old style.